<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Widget</class>
 <widget class="QWidget" name="Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>700</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>TCP Client and Server</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QStackedWidget" name="stackedWidget">
     <widget class="QWidget" name="loginPage">
      <widget class="QLineEdit" name="usernameInput">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>200</y>
         <width>421</width>
         <height>31</height>
        </rect>
       </property>
       <property name="placeholderText">
        <string>Username</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="passwordInput">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>260</y>
         <width>421</width>
         <height>31</height>
        </rect>
       </property>
       <property name="echoMode">
        <enum>QLineEdit::Password</enum>
       </property>
       <property name="placeholderText">
        <string>Password</string>
       </property>
      </widget>
      <widget class="QPushButton" name="loginButton">
       <property name="geometry">
        <rect>
         <x>460</x>
         <y>350</y>
         <width>80</width>
         <height>24</height>
        </rect>
       </property>
       <property name="text">
        <string>Login</string>
       </property>
      </widget>
      <widget class="QLabel" name="label">
       <property name="geometry">
        <rect>
         <x>220</x>
         <y>90</y>
         <width>301</width>
         <height>81</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>26</pointsize>
        </font>
       </property>
       <property name="text">
        <string>TCP Client</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_2">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>200</y>
         <width>81</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Username</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_3">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>270</y>
         <width>91</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Password</string>
       </property>
      </widget>
     </widget>
     <widget class="QWidget" name="mainPage">
      <widget class="QLabel" name="label_4">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>20</y>
         <width>101</width>
         <height>41</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Client</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_5">
       <property name="geometry">
        <rect>
         <x>560</x>
         <y>20</y>
         <width>91</width>
         <height>41</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Server</string>
       </property>
      </widget>
      <widget class="QTextEdit" name="clientTextEdit">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>120</y>
         <width>256</width>
         <height>192</height>
        </rect>
       </property>
      </widget>
      <widget class="QLineEdit" name="clientMessageInput">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>330</y>
         <width>108</width>
         <height>23</height>
        </rect>
       </property>
       <property name="placeholderText">
        <string>Message</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="portInput">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>50</y>
         <width>108</width>
         <height>23</height>
        </rect>
       </property>
       <property name="placeholderText">
        <string>Port</string>
       </property>
      </widget>
      <widget class="QPushButton" name="clientConnectButton">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>80</y>
         <width>141</width>
         <height>24</height>
        </rect>
       </property>
       <property name="text">
        <string>Connect as Client</string>
       </property>
      </widget>
      <widget class="QPushButton" name="clientSendButton">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>360</y>
         <width>80</width>
         <height>24</height>
        </rect>
       </property>
       <property name="text">
        <string>Send</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="addressInput">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>108</width>
         <height>23</height>
        </rect>
       </property>
       <property name="placeholderText">
        <string>Server Address</string>
       </property>
      </widget>
      <widget class="QTextEdit" name="serverTextEdit">
       <property name="geometry">
        <rect>
         <x>400</x>
         <y>120</y>
         <width>256</width>
         <height>192</height>
        </rect>
       </property>
      </widget>
      <widget class="QPushButton" name="serverStartButton">
       <property name="geometry">
        <rect>
         <x>400</x>
         <y>90</y>
         <width>80</width>
         <height>24</height>
        </rect>
       </property>
       <property name="text">
        <string>Start Server</string>
       </property>
      </widget>
      <widget class="QPushButton" name="serverSendButton">
       <property name="geometry">
        <rect>
         <x>400</x>
         <y>370</y>
         <width>80</width>
         <height>24</height>
        </rect>
       </property>
       <property name="text">
        <string>Send</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="serverMessageInput">
       <property name="geometry">
        <rect>
         <x>400</x>
         <y>330</y>
         <width>121</width>
         <height>31</height>
        </rect>
       </property>
       <property name="placeholderText">
        <string>Message</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="serverPortInput">
       <property name="geometry">
        <rect>
         <x>400</x>
         <y>40</y>
         <width>108</width>
         <height>23</height>
        </rect>
       </property>
       <property name="placeholderText">
        <string>Server Port</string>
       </property>
      </widget>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
